client/grpc_client.go

```diff
@@ -217,87 +240,115 @@ func (cli *grpcClient) finishAsyncCall(req *types.Request, res *types.Response)
 	return reqres
 }
 
+func (cli *grpcClient) checkErrGetResult() *types.Result {
+	if cli.err != nil {
+		errorLog := cli.err.Error()
+		cli.StopForError(cli.err)
+		result := types.ErrInternalError
+		result.SetLog(errorLog)
+		return &result
+	}
+	return nil
+}
+
+func (cli *grpcClient) checkGetErr() error {
+	if cli.err != nil {
+		err := errors.New(cli.err.Error())
```

Comment:
oh you mean why copy the error. For some reason I thought the err could change under us, but it shouldnt